### PR TITLE
Improvement: give option to refresh namespace dropdown

### PR DESCRIPTION
**Pull request by Monkeychip**
Issue: When you add a namespace through the CLI, the new addition does not immediately appear in the namespace dropdown on the UI, until you either refresh the page or navigate around.  It was requested that we give an option to refresh the namespace list in the dropdown.  

While we were in there, design requested a couple styling changes on the menu. 

Here is a gif showing how when you add a namespace on the CLI you can use the "Manage namespaces" button to refresh the list.

Here is a gif of just the new menu look:

### CODE REVIEW

**Comment by joshuaogle** [comment]
Looks great! Would it be much more difficult to have the scroll on `namespace-list` rather than the whole container?

On one hand, for people with dozens of namespaces it might make it easier to navigate without losing the current and manage sections, but on the other hand our other dropdowns scroll like what you have here. What do you think?

**Comment by Monkeychip** [comment]
@joshuaogle I agree it might be more helpful to just scroll the namespace list, but I think that would look and feel a bit off.  The scroll bar would be _inside_ the namespace list and that to me feels strange.  I go back and forth, but I think I would ultimately vote for what I would expect to be which is how it is, though the other way might be more helpful.  Thoughts?

**Comment by joshuaogle** [comment]
Ok, let's keep what you have here then. I agree it's a tradeoff and if it's not a clear improvement then consistency wins. 👍 Thanks!

**Comment by meirish** [comment]
Would you ever want to show the refresh if a user didn't have list / manage permissions?

File: ui/app/components/namespace-picker.js [hunk 1]
```diff
@@ -157,4 +157,10 @@ export default Component.extend({
     let parts = namespace.split('/');
     return parts[parts.length - 1];
   }),
+
+  actions: {
+    refreshNamespaceList() {
+      this.get('namespaceService.findNamespacesForUser').perform();
```
Comment (chelshaw) [inline ui/app/components/namespace-picker.js]: Do we need to provide any params here to make sure the result we fetch isn't cached?

Comment (Monkeychip) [reply ui/app/components/namespace-picker.js]: Good question.  I see `.perform()` used throughout the codebase without any params so I am guessing no, we're just calling the method.

**Comment by Monkeychip** [comment]
> Would you ever want to show the refresh if a user didn't have list / manage permissions?

Good thought.  Let me play around with this, but I'll be there's a situation like the one you mentioned where we may not want to show the refresh option.

**Comment by Monkeychip** [comment]
@meirish , follow up question on your comment.  I am trying to figure out a policy that would show the dropdown, and from there I can conditionally hide the refresh button.  However, I haven't been able to create a policy or associated token that will do that.  Even if I have a token with full access to `sys/namespaces/*` the dropdown doesn't show because the property `accessibleNamespaces = null`.  Any thoughts?

**Comment by meirish** [comment]
:wave: @Monkeychip ! The situation I was thinking about was when they have access to read something inside a namespace, but can't manage the namespaces themselves, so "manage namespaces" wouldn't show up, but they still may want the refresh (though this is likely a super edge case). The endpoint used to list namespaces for the menu dynamically returns namespaces that the token has permissions in unlike the `sys/namespaces` endpoint which follows the normal ACL rules.

I _think_ this is a representative example of the situation I was thinking of: 
With a namespace of `ns1` with a secret engine mounted at `ns1/secrets`, and then they have a policy that is something like:
```
#list and read secrets one namespace down
path "+/secrets*" {
  capabilities = ["read", "list"]
}
```

If an admin then added a namespace `ns2/` and enabled a secret engine at `ns2/secrets`, a refresh should then show that user the `ns1` and `ns2` namespace in the list.

**Comment by Monkeychip** [comment]
> ["read", "list"]

@meirish Helpful info, thank you.  I tried out what you were suggesting, and it worked as expected, e.g. the namespaces are correctly listed.  However, the larger problem with conditionally showing or hiding the "Manage namespace" option, is how we are checking capabilities.  Right now, there is a check for `{{#if canList}}` wrapped around "Manage namespaces".  So if they don't have a "list" policy for `sys/namespaces`, they shouldn't see the "Manage namespace" option.  

However, if you have a parent and child namespace, the capabilities check seems to occur at the parent level, and not the child.  Let's say you have a parent namespace called Education, and you have full namespace capabilities, and then you have access to the education/training, with no namespace capabilities. 
```
#namespace education
path "sys/namespaces/*" {
   capabilities = ["create", "read", "update", "delete", "list", "sudo"]
}
#namespace education/training
no capabilities under "sys/namespace"
```

In this situation, `{{#if canList}}` fails because it appears to be checking the parent namespace (e.g. education) permissions and not the child (e.g. education/training).  I suspect the fix for this might require some refactoring, which maybe outside the scope of this ticket. But if you have any thoughts or want to zoom through it, let me know.  Here's a gif showing this specific situation.

**Comment by Monkeychip** [comment]
As a follow up, confirmed with Matthew that this new approach to set `canList` in the `fetchListCapabilities` is good.

**Pull request merged**